go/client/simplefs_ls.go

```diff
@@ -490,85 +491,76 @@ func (c *CmdSimpleFSList) createListing(dirname string, fip FileInfoPath) (Listi
 }
 
 // Comparison function used for sorting Listings by name.
-func compareName(a, b Listing) int {
+func lessListingByName(listings []Listing, i, j int) bool {
+	a := listings[i]
+	b := listings[j]
 	aNameLower := strings.ToLower(a.name)
 	bNameLower := strings.ToLower(b.name)
 
 	var smallerLen int
-	if len(a.name) < len(b.name) {
-		smallerLen = len(a.name)
+	if len(aNameLower) < len(bNameLower) {
```

Comment:
Seems like we can just do this from here to avoid iterating explicitly and explicit check on length:

```go
if aNameLower != bNameLower {
    return aNameLower < bNameLower
}
return a.name >= b.name
```

Am I missing something?

Reply:
Whoa you're right! String comparison works directly in Go!
https://play.golang.org/p/35YxpBwWtsD

But @songgao: did you mean:
```go
if len(aNameLower) != len(bNameLower) {
    return aNameLower < bNameLower
}
```
Since you don't want to interleave uppercase and lowercase words that are the same length.

Reply:
Yeah, this seems fine to me.

Reply:
Also: you can get around the length issue by converting to a rune slice first:
```go
fmt.Println(len([]rune("i")) == len([]rune("İ"))) // prints "true"
```

Reply:
Hmm I still don't understand what the length issue is. The `strings.ToLower` function deals with utf-8 properly and only does the naive iteration (as an optimization) if it's sure the string is all ascii: https://github.com/golang/go/blob/1fadbc1a76b2d9a11453ef4e21b77c353493ac2c/src/strings/strings.go#L593

So the string length shouldn't change right?

Reply:
Oops just saw Fred's comment below. NVM!

Reply:
Oh actually ignore my NVM comment. I don't think that applies to Go's `strings` package. @akalin

Reply:
Yeah that's why I pointed out that casting to a rune slice gets you the "right" length.

Reply:
@jzila hmm but string not equal already implies string length not equal right?

Reply:
Oh didn't see your last comment @jzila . Out curiosity I found this too: https://golang.org/pkg/unicode/utf8/#RuneCountInString

Reply:
Nice bike shed you all built here.  My original goal here was to change as little of the original code as possible.  But I'll go with song's original suggestion, no need to bring runes into this.

Reply:
Achievement Unlocked: `built a bike shed`